### ISSUE TITLE
Störungsmeldungen - Findings Roger

### ISSUE THREAD

**Issue by TO-mdv** [openTdataCH/ojp-sdk#156]
https://docs.google.com/document/d/1Jl3KN0f3FBedyNKJQ2Rx6pdb1gWMjd3LjAm7S5zQd6o/edit?tab=t.0
[SH-3959](https://youtrack.mentz.net/issue/SH-3959) OJP2.0 Ausgabe Störungsmeldungen anpassen - Findigs Roger Kneubühl

**Anpassungen**

- Inhalt von /PtSituation/ScopeType auch für Element /PtSituation/PublishingActions/PublishingAction/PublishAtScope/ScopeType übernehmen

- Komplette Struktur entfernen /PtSituation/Affects

- Komplette Struktur /PtSituation/PublishingActions/PublishingAction/PublishAtScope/Affects wie folgt leer ausgeben: <siri:Affects></siri:Affects>

- Komplette Struktur entfernen /PtSituation/Consequences

- Element <siri:UnknownReason>unknown</siri:UnknownReason> entfernen. Es muss jedoch sichergestellt sein, dass Element /PtSituation/AlertCause immer ausgegeben wird (eines davon muss vorhanden sein)

- Hardcodierter Wert von /PtSituation/siri:Source/siri:SourceType ändern von other nach directReport

**Comment by r3to** [comment]
**Konsequenzen der Findings für die Mobile SDKs:**

- Das [Affects-Element](https://vdvde.github.io/OJP/develop/documentation-tables/siri.html#type_siri__AffectsScopeStructure) haben wir nicht vollständig modelliert. OJP liefert mehr Informationen, als wir im SDK abgebildet haben.
- Affects kann sowohl auf `PTSituation` (optional) wie auf `PTSituation > PublishingActions ... > PublishAtScope` (required) sein. Wir haben es nur auf `PTSituation`

**Comment by TO-mdv** [comment]
@r3to wir haben nun eine angepasste, den Vorschläge von Roger folgenden Ausgabe der Störungsmeldungen umgesetzt. Du könntest bei Gelegenheit auf INT testen, den Release auf PROD stimmen wir gemeinsam ab.

**Comment by r3to** [comment]
@deka91 und @nicoti8m könnt ihr prüfen, ob auf dem Android SDK auch was angepasst werden muss?
Für das iOS SDK scheint es keine Breaking Changes gegeben zu haben. Haben jetzt nur "PublishAtScope" hinzugefügt und Affects von PTSituation entfernt:

https://github.com/openTdataCH/ojp-ios/pull/92/files#diff-36a3d4bf03b34d093ef73d0d0b39c00b1ce9808104dfb7c69c329640b8d5b3d7R215-R265

**Comment by deka91** [comment]
@r3to Bei uns gibt es ebenfalls keine Breaking Changes. Die Code-Anpassungen habe ich im SDK gemacht und wir würden dann gemeinsam in Abstimmung einen neuen Release planen.

**Comment by TO-mdv** [comment]
@r3to @deka91 Das heisst für mich konkret, ich muss mit dem go-live dieser Änderung noch warten, bis ihr das SDK released habt? Wann plant ihr diesen Release ?

**Comment by r3to** [comment]
@TO-mdv nein, wir haben bis jetzt keine Breaking Changes entdeckt. Das bestehende SDK scheint auch mit den Änderungen auf INT gut zurecht zu kommen, weil die betroffenen Elemente sowieso Optional waren. Sollte dem Release eigentlich nichts im Weg stehen
